.github/workflows/test.yml

```diff
@@ -0,0 +1,56 @@
+name: Testing
+
+on:
+  pull_request:
```

Comment:
@lpirola analisando especificamente o momento de executar os testes fico pensando se faz sentido executar em todos os commits, dessa forma conseguimos identificar o problema de forma mais rápida.  Vou mandar aqui alguns pontos positivos e negativos baseado das duas abordagens:

1. **Execução dos Testes em Cada Commit**

Vantagens:
- Detecta erros e regressões rapidamente, proporcionando feedback imediato ao desenvolvedor.
- Garante que o código na branch esteja sempre em um estado funcional.

Desvantagens:
- Pode ser demorado se o conjunto de testes for grande, afetando a produtividade.
- Pode ser excessivo em projetos menores ou com commits muito frequentes.

2. **Execução dos Testes no Merge Request (Pull Request) para a develop**

Vantagens:
- Permite revisão de código com confiança de que as alterações não quebram funcionalidades existentes.
- Centraliza a validação do código em momentos cruciais, como a integração de novas funcionalidades.

Desvantagens:
- O feedback não é tão imediato quanto nos commits.
- Problemas podem ser detectados tardiamente, após várias alterações já terem sido feitas.

Para adicionar em todos os commits podemos usar:

```
on: 
  push:
    # only trigger on branches, not on tags
    branches: '**'
```

Reply:
feito!